Tools/ros2/README.md

```diff
@@ -0,0 +1,244 @@
+# ArduPilot ROS 2 packages
+
+ This directory contains ROS 2 packages and configuration files for running
+ ROS 2 processes and nodes that communicate with the ArduPilot DDS client
+ library using the microROS agent. It contains the following packages:
+ 
+#### `ardupilot_sitl`
+
+A `colcon` package for building and running ArduPilot SITL using the ROS 2 CLI.
+For example `ardurover` SITL may be launched with:
+
+```bash
+ros2 launch ardupilot_sitl sitl.launch.py command:=ardurover model:=rover
+```
+
+#### `ardupilot_dds_test`
+
+A `colcon` package for testing communication between `micro_ros_agent` and the
+ArduPilot `AP_DDS` client library.
+
+## Prerequisites
+
+The packages depend on:
+
+- [ROS 2 Humble](https://docs.ros.org/en/humble/Installation/Ubuntu-Install-Debians.html)
+
+
+## Install Ubuntu
+
+#### 1. Create a workspace folder
+
+```bash
+mkdir -p ~/ros_ws/src && cd ~/ros_ws/src
+```
+
+The ROS 2 tutorials contain more details regarding [ROS 2 workspaces](https://docs.ros.org/en/humble/Tutorials/Workspace/Creating-A-Workspace.html).
+
+#### 2. Get the `ros2.repos` file
+
+```bash
+cd ~/ros2_ws/src
+wget https://raw.githubusercontent.com/srmainwaring/ardupilot/pr/pr-dds-launch-tests/Tools/ros2/ros2.repos
+vcs import --recursive < ros2.repos
+```
+
+#### 3. Update dependencies
+
+```bash
+cd ~/ros_ws
+source /opt/ros/humble/setup.bash
+sudo apt update
+rosdep update
+rosdep install --rosdistro humble --from-paths src
+```
+
+#### 4. Build
+
+```bash
+cd ~/ros_ws
+colcon build --packages-select micro_ros_agent
+colcon build --packages-select ardupilot_sitl
+colcon build --packages-select ardupilot_dds_tests
+```
+
+#### 5. Test
+
+```bash
+source ./install/setup.bash
+colcon test --pytest-args -s -v --event-handlers console_cohesion+ --packages-select ardupilot_dds_tests
+colcon test-result --all --verbose
```

Comment:
Have you considered also hooking in to the Junit output of colcon test so the tests show up in Gitlab?
https://github.com/marketplace/actions/junit-report-action

Reply:
Not looked into this. Sounds like a good idea for the GHW.

Reply:
You can cover it in a follow up PR if you like. Looking at test reports in GH UI can be nicer than scrolling through long logs.

Reply:
Yes, let's do that.